examples/crowdfunding/assets/crowdFund.teal

```diff
@@ -0,0 +1,469 @@
+// this parameter should be addr:
+byte "Escrow"
+txna ApplicationArgs 0
+app_global_put
+int 1
+return
+
+not_update:
+// check for closeout
+int CloseOut
+txn OnCompletion
+==
+bnz close_out
+
+// check if no params are
+// passed in, which should
+// only happen when someone just
+// wants to optin
+// note the code is written
+// to allow opting in and donating with
+// one call
+int 0
+txn NumAppArgs
+==
+bz check_parms
+
+// Verify someone is 
+// not just opting in
+int OptIn
+txn OnCompletion
+==
+bz failed
+int 1
+return
+
+check_parms:
+// donate
+txna ApplicationArgs 0
+byte "donate"
+==
+bnz donate
+
+// reclaim
+txna ApplicationArgs 0
+byte "reclaim" 
+==
+bnz reclaim
+
+// claim
+txna ApplicationArgs 0
+byte "claim" 
+==
+bnz claim
+b failed
+
+
+donate:
+// check dates to verify
+// in valid range
+// Note: Timestamp check is excluded from this example, 
+// To add it, remove comments from following lines.
+// global LatestTimestamp
+// byte "StartDate"
+// app_global_get
+// >=
+// global LatestTimestamp
+// byte "EndDate"
+// app_global_get
+// <=
+// &&
+// bz failed
+
+// check if grouped with
+// two transactions
+global GroupSize
+int 2
+==
+
+// second tx is an payment
+gtxn 1 TypeEnum
+int 1
+==
+&&
+bz failed
+
+// verify escrow is receiving 
+// second payment tx
+byte "Escrow"
+app_global_get
+gtxn 1 Receiver
+==
+bz failed
+
+// increment the total 
+// funds raised so far
+byte "Total"
+app_global_get
+gtxn 1 Amount
++
+store 1
+byte "Total"
+load 1
+app_global_put
+
+// increment or set giving amount
+// for the account that is donating
+int 0 //sender
+txn ApplicationID
+byte "MyAmountGiven"
+app_local_get_ex
+
+// check if a new giver
+// or existing giver
+// and store the value
+// in the givers local storage
+bz new_giver
+gtxn 1 Amount
++
+store 3
+int 0 //sender
+byte "MyAmountGiven"
+load 3
+app_local_put
+b finished
+new_giver:
+int 0 //sender
+byte "MyAmountGiven"
+gtxn 1 Amount
+app_local_put
+b finished
+
+
+claim:
+// verify there are 2 transactions
+// in the group
+global GroupSize
+int 2
+==
+bz failed
+
+// verify that the receiver 
+// of the payment transaction
+// is the address stored 
+// when the fund was created
+gtxn 1 Receiver
+byte "Receiver"
+app_global_get
+==
+
+// verify the sender
+// of the payment transaction
+// is the escrow account
+gtxn 1 Sender
+byte "Escrow"
+app_global_get
+==
+&&
+
+// verify that the CloseRemainderTo
+// attribute is set to the receiver
+gtxn 1 CloseRemainderTo 
+byte "Receiver"
+app_global_get
+==
+&&
+
+// verify that the fund end date 
+// has passed
+// global LatestTimestamp
+// byte "EndDate"
+// app_global_get
+// >
+// &&
+bz failed
+
+// verify that the goal was reached
+byte "Total"
+app_global_get
+byte "Goal"
+app_global_get
+>=
+bz failed
+b finished
+
+//dont check amount because
+//escrow account may receiver 
+//more than the goal
+//Could check for total vs tx1 amount
+
+//Fund did not meet total
+//Allow ppl to reclaim funds
```

Comment:
people?

Reply:
yes, i think now we have to be very careful while reading from algorand documentation and tutorial. there are some logical , and many minor errors.